.github/workflows/workflow-autoapply.yml

```diff
@@ -0,0 +1,183 @@
+  AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+
+jobs:
+  plan_apply:
+    name: Plan and Apply
+    runs-on: [self-hosted, Linux, AWS]
+    timeout-minutes: 60
+
+    environment: ${{ github.event_name != 'pull_request' && matrix.environment || null }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: ${{ fromJson(inputs.config).*.workspaces.*.environment }}
+
+    steps:
+    - id: parse
+      uses: Brightspace/terraform-workflows/actions/parse-config@v4
+      with:
+        # GHA's fromJson() supports comments in JSON strings which is nice from
+        # an end-user perspective (e.g. naming account ids). toJson(fromJson())
+        # is used in order to strip these comments out before parsing it ourselves
+        config: ${{ toJson(fromJson(inputs.config)) }}
+        max_environments: 50
+
+    - uses: Brightspace/third-party-actions@actions/checkout
+
+    - if: ${{ github.event_name != 'pull_request' }}
+      name: 'Assert: Branch is up to date'
+      uses: Brightspace/assert-git-remote-ref-action@main
+      with:
+        remote: origin
+        ref: ${{ inputs.default_branch }}
+
+    - if: ${{ inputs.artifacts_url != '' }}
+      name: 'Extract artifacts'
+      uses: Brightspace/s3-artifact-actions/extract@master
+      with:
+        path: ${{ fromJson(steps.parse.outputs.config)[matrix.environment].workspace_path }}/.artifacts
+        artifacts_url: ${{ inputs.artifacts_url }}
+
+    - id: plan
+      uses: Brightspace/terraform-workflows/actions/plan@v4
+      with:
+        comment_plan: ${{ inputs.comment_plan }}
+        config: ${{ toJson(fromJson(steps.parse.outputs.config)[matrix.environment]) }}
+        terraform_version: ${{ inputs.terraform_version }}
+        refresh_on_pr: ${{ inputs.refresh_on_pr }}
+        skip_archive: true
+
+    # Matrixed jobs overwrite eachother's outputs if they use the same name.
+    # Unfortunately though, job outputs need to be statically defined in the yml
+    # Step outputs are dynamic though, as well as the _population_ of job outputs
+    # (based on the existence of their configured value). This step uniquely maps
+    # the environment name to one of the statically defined outputs below. The
+    # apply job can then consume the sparse set of outputs that were assigned a
+    # value via `needs.plan.outputs.*`
+    - if: ${{ steps.plan.outputs.has_changes == 'true' }}
+      name: Mark as changed
+      id: changed
+      shell: bash
+      run: |
+        ENVIRONMENT_INDEX=$(jq -cr \
+          --arg environment "${ENVIRONMENT}" \
+          '
+            to_entries[]
+            | select(.value == $environment)
+            | .key
+          ' \
+          <<< "${ENVIRONMENTS}"
+        )
+        echo "changed_env_${ENVIRONMENT_INDEX}=${ENVIRONMENT}" >> "${GITHUB_OUTPUT}"
+      env:
+        ENVIRONMENTS: ${{ steps.parse.outputs.environments }}
+        ENVIRONMENT: ${{ matrix.environment }}
```

Comment:
This stuff is still here so the workflow can still output `had_changes`. repo-settings consumes that to decide whether to dispatch to downstream jobs